### PR TITLE
feat: add finalize round component & upload matching JSON

### LINKED ISSUE #867: Add Finalize round toggle
**User Story:**
As a round operator
I want to finalize and store my fund distribution
So that my round results are public and verifiable

**Acceptance Criteria**
GIVEN that I am on the Funding Admin tab after my round has ended
WHEN I look below the Download JSON button
THEN I see a toggle to select default or upload my own json
AND the default radial button is selected

GIVEN that I am on the Funding Admin tab after my round has ended
WHEN I click on the upload my own radial button
THEN I see the upload section

GIVEN that I am on the Funding Admin tab after my round has ended and have Upload My Own selected
WHEN I click on the default radial button
THEN the Upload section disappears

**Product & Design Links:**
[Figma](https://www.figma.com/file/K31Tf7tSn10BuN65VC6YLc/Round-Manager---Hi-Fidelity-Designs?node-id=3519%3A49180&t=cdEqlYuPoK2gd0TW-4)

**Tech Details:**

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### LINKED ISSUE #868: Upload JSON matching file
**User Story:**
As a round operator
I want to finalize and store my fund distribution
So that my round results are public and verifiable

**Acceptance Criteria**

GIVEN that I am on the Funding Admin tab in the Upload My Own state
WHEN I click on the upload JSON button
THEN I am prompted to select a file to upload

GIVEN that I am on the Funding Admin tab and have selected a file to upload
WHEN I click to upload
THEN I see the "Heads Up" warning

GIVEN that I am on the Funding Admin tab and can see the Heads Up warning
WHEN I click Continue
THEN my file is uploaded

GIVEN that I am on the Funding Admin tab and can see the Heads Up warning
WHEN I click Cancel
THEN my file upload is cancelled
AND the tab reverts to the Upload my own state

GIVEN that I am on the Funding Admin tab
WHEN I upload a JSON
THEN I see the User Uploaded matching stats

GIVEN that I am on the Funding Admin tab
WHEN I upload a JSON **with percentages that don't add up to 100%** 
THEN I see an error message that says "Matching Percentages must add up to 100%"

**Product & Design Links:**
[Figma](https://www.figma.com/file/K31Tf7tSn10BuN65VC6YLc/Round-Manager---Hi-Fidelity-Designs?node-id=3342%3A38901&t=AJekSKSzpyWU5TFj-4)

**Tech Details:**

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### LINKED ISSUE #869: Save matching percentage to contract
**User Story:**
As a round operator
I want to finalize and store my fund distribution
So that my round results are public and verifiable

**Acceptance Criteria**
GIVEN that I am on the Funding Admin tab after my round has ended
WHEN I click on Save to Contract
THEN I am prompted to sign a transaction

GIVEN that I am on the Funding Admin tab and am looking at my wallet transaction
WHEN I click confirm
THEN a transaction to save my selected matching stats (default or my own) is queued
AND I see the Processing modal

GIVEN that I am on the Funding Admin tab and am looking at the processing modal
WHEN I the transaction completes
THEN I return to the default state of the Funding Admin tab

**Product & Design Links:**
[Figma](https://www.figma.com/file/K31Tf7tSn10BuN65VC6YLc/Round-Manager---Hi-Fidelity-Designs?node-id=3342%3A38901&t=AJekSKSzpyWU5TFj-4)

**Tech Details:**

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### PR DESCRIPTION
**Pull request by hmrtn**
As a round operator
I want to finalize and store my fund distribution
So that my round results are public and verifiable

fixes #867 
fixes #868

### CODE REVIEW

**Comment by hmrtn** [comment]
Note: The tests for JSON upload have been commented out because of jest. This error occurs when Jest tries to redefine the TextDecoder property, which is a read-only property in JavaScript.

File: packages/round-manager/src/features/round/__tests__/ViewRoundPage.test.tsx [hunk 1]
```diff
@@ -312,8 +312,233 @@ describe("View Round", () => {
...
+  //
+  //   expect(mockFile.arrayBuffer).toHaveBeenCalled();
+  //
+  //   // expect the table to not be rendered
+  //   await waitFor(() => {
+  //     expect(
+  //       screen.queryByTestId("matching-stats-table")
+  //     ).not.toBeInTheDocument();
+  //   });
+  // });
```
Comment (thelostone-mc) [inline packages/round-manager/src/features/round/__tests__/ViewRoundPage.test.tsx:543]: ^ shouldn't tests be in ViewAdminFundingPage.test ?

File: packages/round-manager/src/features/round/__tests__/ViewFundingAdmin.test.tsx [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+    expect(screen.getByTestId("match-stats-title")).toBeInTheDocument();
+    expect(screen.getByTestId("matching-stats-table")).toBeInTheDocument();
+    expect(screen.getByTestId("custom-or-default-test-id")).toBeInTheDocument();
+  
+    const customRadioButton = screen.getByTestId("custom-radio-test-id");
+    fireEvent.click(customRadioButton);
+    expect(screen.getByTestId("dropzone")).toBeInTheDocument();
+
+    // test dropzone .json file upload
+    const dropzone = screen.getByTestId("dropzone");
```
Comment (thelostone-mc) [inline packages/round-manager/src/features/round/__tests__/ViewFundingAdmin.test.tsx:333]: TODO: this piece fails. I think it's how we've set the mockFile but I'm not sure

File: packages/round-manager/src/features/round/__tests__/ViewRoundPage.test.tsx (hunk 1 above)
Comment (hmrtn) [reply packages/round-manager/src/features/round/__tests__/ViewRoundPage.test.tsx]: They can be - but it is a subcomponent of the view round page. So loading, clicking tabs, for other tests is also in the view round tests.

File: packages/round-manager/src/features/round/__tests__/ViewFundingAdmin.test.tsx (hunk 2 above)
Comment (hmrtn) [reply packages/round-manager/src/features/round/__tests__/ViewFundingAdmin.test.tsx:347]: resolved by mocking the text decoder

**Comment by vacekj** [comment]
If I understand it correctly, the actual encoding and uploading of the matching data to the contract needs to be handled here (I just included a commented-out skeleton) to satisfy all AC in #867. I also see that this branches is named after 869 - should that also be in this PR? I'm confused about the ACs this PR tries to satisfy.

**Review by thelostone-mc — APPROVED** [review]

**Comment by melissa-neira** [comment]
I'm stuck with this loading screen 😵‍💫

Screenshot 2023-01-05 at 12 55 44

**Review by melissa-neira — APPROVED** [review]

**Pull request merged**